app/static/js/bonnerManagement.js

```diff
@@ -14,14 +14,21 @@ function cohortRequest(year, method, username){
   })
 }
 
+function addSearchCapabilities(inputElement){
+    inputElement = $(inputElement);
+    inputElement.on("input", function(){
+        let year = $(this).data('year');
+        searchUser(this.id, s => cohortRequest(year, "add", s.username), false, null, "student");
+    });
+}
+
+
 /*** Run After Page Load *************************************/
 $(document).ready(function(){
-    $("input[type=search]").on("input", function(){
-        let year = $(this).data('year')
-        searchUser(this.id, s => cohortRequest(year, "add", s.username), false, null, "student")
-    });
+    $("#addCohort").on('click', addCohort);
+    $("input[type=search]").each((i, e) => addSearchCapabilities(e));
```

Comment:
It would be great if we get more descriptive variable names here and across the board for each's and maps's. e, s, and element can often be a little confusing.

Reply:
Gotcha